torch/distributed/elastic/timer/file_based_local_timer.py

```diff
@@ -68,69 +64,67 @@ class FileTimerClient(TimerClient):
     Client side of ``FileTimerServer``. This client is meant to be used
     on the same host that the ``FileTimerServer`` is running on and uses
     pid to uniquely identify a worker.
-    This client uses a named_pipe to send timer requests to the
+    This client uses a temp file to send timer requests to the
     ``FileTimerServer``. This client is a producer while the
     ``FileTimerServer`` is a consumer. Multiple clients can work with
     the same ``FileTimerServer``.
 
     Args:
 
-        file_path: str, the path of a FIFO special file. ``FileTimerServer``
-                        must have created it by calling os.mkfifo().
+        file_path: str, the path of a watchdog file.
+
+        timeout_sec: int, watchdog timeout.
+
+        scope_id: str, scope ID of worker
 
         signal: signal, the signal to use to kill the process. Using a
                         negative or zero signal will not kill the process.
     """
-    def __init__(self, file_path: str, signal=(signal.SIGKILL if sys.platform != "win32" else
-                                               signal.CTRL_C_EVENT)) -> None:  # type: ignore[attr-defined]
+    def __init__(
+        self,
+        file_path: str,
+        timeout_sec: int,
+        scope_id: str,
+        signal=(signal.SIGKILL if sys.platform != "win32" else
+                signal.CTRL_C_EVENT)) -> None:  # type: ignore[attr-defined]
         super().__init__()
         self._file_path = file_path
+        self._timeout_sec = timeout_sec
         self.signal = signal
-
-    def _open_non_blocking(self) -> Optional[io.TextIOWrapper]:
-        try:
-            fd = os.open(self._file_path, os.O_WRONLY | os.O_NONBLOCK)
-            return os.fdopen(fd, "wt")
-        except Exception:
-            return None
-
-    def _send_request(self, request: FileTimerRequest) -> None:
-        # The server may have crashed or may haven't started yet.
-        # In such case, calling open() in blocking model blocks the client.
-        # To avoid such issue, open it in non-blocking mode, and an OSError will
-        # be raised if the server is not there.
-        file = self._open_non_blocking()
-        if file is None:
-            raise BrokenPipeError("Could not send the FileTimerRequest because FileTimerServer is not available.")
-        with file:
+        log.info(
+            "Starting %s..."
+            " file_path=%s,"
+            " timeout=%s,"
+            " signal=%s",
+            type(self).__name__,
+            self._file_path,
+            self._timeout_sec,
+            self.signal,
+        )
+        with open(self._file_path, "w") as file:
+            # FileTimerRequest at init is used to write client level
+            # metadata once to watchdog file, later acquire will not
+            # write duplicate data
+            request = FileTimerRequest(
+                worker_pid=os.getpid(),
+                scope_id=scope_id,
+                expiration_time=timeout_sec,
```

Comment:
should we rename this field to be timeout instead of expiration?

Reply:
FileTimerRequest is derived from TimerRequest which has slightly different use model and uses expiration time. that's we are fitting timeout into expiration.

one option can be to have empty expiration_time & create new timeout field.